azure-iot-device/tests/iothub/client_fixtures.py

```diff
@@ -71,4 +67,15 @@ def send_method_response(self, method_response, callback=None):
 
 @pytest.fixture
 def pipeline(mocker):
+    """This fixture will automatically handle callbacks and should be
+    used in the majority of tests.
+    """
     return mocker.MagicMock(wraps=FakePipelineAdapter())
+
+
+@pytest.fixture
+def pipeline_manual_cb(mocker):
+    """This fixture is for use in tests where manual triggering of a
+    callback is required
+    """
+    return mocker.MagicMock()
```

Comment:
>() [](start = 27, length = 2)

spec=FakePipelineAdapter()?

Reply:
Nope. I mean, I could, that would also work, but there's no real reason to that I can see (unless we want to be extremely rigid about specing Mocks whenever possible - And even if we did, the more correct thing to do would be to spec on the actual PipelineAdapter)

---
In reply to: [290983218](https://github.com/Azure/azure-iot-sdk-python-preview/pull/101#discussion_r290983218) [](ancestors = 290983218)